js/ts/chocolatey-tab-multiples.ts

```diff
@@ -0,0 +1,67 @@
+import { Tab } from 'bootstrap';
+import { getCookie, setCookieExpirationNever } from './util/chocolatey-functions';
+
+(() => {
+    const tabMultiAttribute = 'data-choco-tab-multi';
+    const tabMultiElements = document.querySelectorAll<HTMLElement>(`[${tabMultiAttribute}]`);
+    const tabCookies = new Set<string>(); // Use a Set to store unique cookie names
+
+    for (const tabElement of tabMultiElements) {
+        const tabMultiConfigAttribute = tabElement.getAttribute(tabMultiAttribute).replace(/\s/g, '');
+        let tabMultiConfig: { [key: string]: string } | null = null;

```

Comment:
Am I reading this line correctly: We're instantiating a variable called `tabMultiConfig`, and it has a type of `HashTable` (in PowerShell at least, C# would likely call it a Dictionary), or the type could also be `null`. And we're initializing it with `null`.

Reply:
Yes, pretty much. So in normal JS, we'd just define it as `let tabMultiConfig`, but in TypeScript we need to tell it what types it can accept eventually. In this case it can accept an `obj` with the specific structure, or it could be `null`. 

Later down on lines 14-19 I am reassigning the `tabMultiConfig` variable with the contained JSON inside of the `tabElement` html element's attributes. I then test the JSON to ensure it's valid. If it is not in the structure specified, or if it is empty, or other syntax errors, it will throw an error and the script will stop.

Reply:
This would have helped so much when I was doing some `vscode-powershell` work and we ended up doing `let whatever: any` since there were a few things it could be. I'm sure I'll completely forget this when I actually need to use it, but at least for now I know about this trick 😁